fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -1867,6 +1900,54 @@ protected void saveStoreHeader(@Nonnull RecordMetaDataProto.DataStoreInfo storeH
         }
     }
 
+    @Nonnull
+    private CompletableFuture<Void> updateStoreHeaderAsync(@Nonnull UnaryOperator<RecordMetaDataProto.DataStoreInfo> storeHeaderOperator) {
+        if (recordStoreStateRef.get() == null) {
+            return preloadRecordStoreStateAsync().thenCompose(vignore -> updateStoreHeaderAsync(storeHeaderOperator));
+        }
+        AtomicReference<RecordMetaDataProto.DataStoreInfo> oldStoreHeaderRef = new AtomicReference<>();
```

Comment:
Sure. They values of these are now set within the `updateAndGet` closure on the `recordStoreStateRef`. Referenced variables in Java lambas must be final or effectively final, so rather than just setting the local variable, it needs to stash the variable in a holder, which is what the `AtomicReference` gets us.

Reply:
Aha, I see. Thanks Alec.